app/controllers/nonresident/CalculatorController.scala

```diff
@@ -122,6 +123,60 @@ trait CalculatorController extends BaseController {
 
     Future.successful(Ok(Json.toJson(result)))
   }
+
+  def useRebasedCalculation(rebasedValue: Option[Double], rebasedCosts: Option[Double], improvementsAft: Option[Double]): Boolean = {
+    Seq(rebasedValue, rebasedCosts, improvementsAft).forall(_.isDefined)
```

Comment:
use proper names `improvementsAft` has no meaning

This suggests if not improvements after the tax start date are not supplied then a rebased gain cannot be calculated. We could default this to 0  in the route and stop it being optional - just a suggestion.